src/bots/enigmabot.js

```diff
@@ -22,6 +22,7 @@ let ai = {
     emptyTerritories: [], // The map locations we can see that are free to conquer.
     foggedTerritories: [],
     visibleOpponentTerritories: [],
+    discoveredCities: [], // List of all cities that have been found during the game.  City may be currently fogged.
```

Comment:
The plan is for the client to do this, so that the functionality is provided to all of the bots, similarly to how we keep a persistent list of general locations, here: https://github.com/skye2k2/generals-bot/blob/master/src/client.js#L241

Reply:
I'll rework it that way.